File: 2º Periodo/AEDS-II/BeeCrowd/Java/quantidade pokemons.java
import java.util.Scanner;

class Celula {

	public String pokemon;
	public Celula prox;



	public Celula() {

		this(" ");



	}


	public Celula(String pokemon) {

		this.pokemon = pokemon;
		this.prox = null;




	}


}

class Lista {

	public Celula primeiro;
	public Celula ultimo;




	public Lista() {


		this.primeiro = new Celula(" ");
		this.ultimo = primeiro;



	}


	public void inserirFim(String x) {

		boolean repetido = false;

		for(Celula i = primeiro.prox; i != null; i = i.prox) {

			if(x.equals(i.pokemon)) {
				repetido = true;
			}

		}

		if(repetido != true) {

			ultimo.prox = new Celula(x);
			ultimo = ultimo.prox;


		}


	}


	public void mostrar() {

        int pokemons = 151;
        int quantidade = 0;
		for(Celula i = primeiro.prox; i != null; i = i.prox) {


			quantidade++;

		}


        pokemons -= quantidade;
        System.out.print("faltam " + pokemons );



	}


}

public class Main
{
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		Lista lista = new Lista();

		int quantidade = scanner.nextInt();
		scanner.nextLine();

		String pokemon;

		for(int i = 0; i <= quantidade; i++) {


			pokemon = scanner.nextLine();

			lista.inserirFim(pokemon);

		}


		lista.mostrar();


	}
}

File: 2º Periodo/AEDS-II/BeeCrowd/C/quantidade pokemons.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

typedef struct Celula {
    
    char pokemon[15];
    struct Celula* prox;
   
}Celula;


Celula* novacelula(char pokemon[]){
    
    Celula* nova = (Celula*) malloc(sizeof(Celula));
    
    strcpy(nova->pokemon, pokemon);
    nova -> prox = NULL;
    return nova;
    
    
}


Celula* primeiro;
Celula* ultimo;


void start(){
    
  primeiro = novacelula(" ");
  ultimo = primeiro;
    
    
}


void inserirFim(char x[] ){
    bool repetido = false;
    
    for(Celula* i  = primeiro-> prox; i != NULL; i = i -> prox){
        
        if(strcmp(x, i->pokemon) == 0){
        repetido = true;
        }
    }
  
  
  if(repetido != true){
    ultimo->prox = novacelula(x);
    ultimo = ultimo->prox;

    
  }
    
}

void mostrar(){
    
  int quant = 0;
   for(Celula* i = primeiro->prox; i != NULL; i = i-> prox){
       
       
       
     quant++;
       
   } 
    
    int pokemons = 151;
    pokemons -= quant;
    printf("Falta(m) %d pokemon(s)", pokemons);
    
}


int main()
{
    
    
    start();
    
    int quantidade;
    scanf("%d" , &quantidade);
    
    char pokemon[15];

    
    for(int i = 0; i < quantidade; i++){
        
       scanf("%s" , pokemon); 
        inserirFim(pokemon);
        
    }
    
    
    mostrar();


    return 0;
}